topology/src/main/java/io/camunda/zeebe/topology/api/TopologyRequestServer.java

```diff
@@ -122,8 +122,14 @@ private CompletableFuture<Either<ErrorResponse, TopologyChangeResponse>> mapResp
       final ActorFuture<TopologyChangeResponse> topologyManagementApi) {
     return topologyManagementApi
         .toCompletableFuture()
-        .thenApply(Either::<ErrorResponse, TopologyChangeResponse>right)
-        .exceptionally(this::mapError);
+        .handle(
+            (response, throwable) -> {
+              if (throwable == null) {
+                return Either.right(response);
+              } else {
+                return mapError(throwable);
+              }
+            });
```

Comment:
:upside_down_face: I didn't know you had fixed this. I pushed a fix to the other branch.